renderlib/Colormap.h

```diff
@@ -60,17 +60,76 @@ struct ColorControlPoint
   }
 };
 
-uint8_t*
-colormapFromControlPoints(std::vector<ColorControlPoint> pts, size_t length = 256);
-uint8_t*
-modifiedGlasbeyColormap(size_t length = 256);
-uint8_t*
-colormapRandomized(size_t length = 256);
-uint8_t*
-colormapFromColormap(uint8_t* colormap, size_t length = 256);
+class ColorRamp
```

Comment:
the changes in this file introduce the ColorRamp class which encapsulates a name, the color gradient stops, and a "baked" or cached version of the colormap as an array / lookup table.